assets/src/models/blockWaiver.ts

```diff
@@ -21,3 +21,6 @@ export const currentFuturePastType = ({
     return CurrentFuturePastType.Current
   }
 }
+
+export const hasBlockWaiver = ({ blockWaivers }: VehicleOrGhost): boolean =>
```

Comment:
I think I understand what you mean by, "more in the way there". In my mind the tension between that and what feels most appropriate from the viewpoint of delegation of responsibility makes me wonder if there is room to drill into this further.

So for example, `hasBlockWaiver` feels pretty directly like a question about a vehicle and its state to me. But maybe `hasCurrentBlockWaiver` is a combination of concerns, like the outer function could live in `Vehicle` and include the `some` over the collection, but for each individual block waiver that could be delegated to a `isCurrent` function on `BlockWaiver` or something along those lines?

And then finally `blockWaiverDecoratorStyle` does feel different to me too. It's a combination concerns and the underlying question is a little less about interrogating the state of a "thing" than about deciding how to present information. So for me I could see that being in `BlockWaiver` since it's conveying information about block waivers, or `Vehicle` because you are deciding how to draw the representation of a vehicle, or even into a place that might be more "view-related" or I guess maybe "view-model-related" than "model-related".

But in the end, take any of that if it's helpful, but your call.

Reply:
Hm, yeah, I see that, and maybe if we had more organization for view-specific code or if we had another use for `isCurrentBlockWaiver`, then those changes would definitely be good. We could discuss it more, but I think for now this is fine and don't want to get hung up on something this small, so I'll leave it as it it is.